### PR TITLE
[CI] Adding ConvectionDiffusionApplication to CI

### PR DESCRIPTION
**Pull request by rubenzorrilla**
As the ConvectionDiffusionApplication is understood as a core application, I think it must be in the CI.

I'd like to add that it has been already proposed in #5530 but we rejected it in favour of TrilinosApplication and MetisApplication because we were tight to the Travis and Appveyor times.

### CODE REVIEW

**Comment by philbucher** [comment]
just one question, how is the process of cleaning up the app?
as far as I remember there was quite some legacy code without testing there

**Comment by philbucher** [comment]
tests are crashing and taking too long

**Comment by rubenzorrilla** [comment]
> tests are crashing and taking too long

I'll have a look.

**Comment by rubenzorrilla** [comment]
> just one question, how is the process of cleaning up the app?
> as far as I remember there was quite some legacy code without testing there

There is a mix between new stuff and legacy stuff. The new features work and have tests.

Legacy code has no tests and possibly not compatible with the current Kratos standards. The point is that, as far as I know, there is still people using it.

**Comment by philbucher** [comment]
tests are still taking too long
I guess you need to remove some tests from the smallsuite

**Comment by philbucher** [comment]
ah now I get why you were asking how to measure the time :)

**Comment by philbucher** [comment]
Seems like you need to replace the ExternalSolversApp in the tests with the EigenSolversApp

File: applications/ConvectionDiffusionApplication/tests/test_ConvectionDiffusionApplication.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import KratosMultiphysics.KratosUnittest as KratosUnittest
 
 try:
-    import KratosMultiphysics.ExternalSolversApplication as ExternalSolversApplication
+    import KratosMultiphysics.EigenSolversApplication as EigenSolversApplication
```
Comment (philbucher) [inline applications/ConvectionDiffusionApplication/tests/test_ConvectionDiffusionApplication.py]: better use Kratos_utilitis for this

also this check is not used, the tests that use the EigenSolversApp should only be run if it is available

Comment (rubenzorrilla) [reply applications/ConvectionDiffusionApplication/tests/test_ConvectionDiffusionApplication.py]: I did a search and replace. Let me check.

Comment (rubenzorrilla) [reply applications/ConvectionDiffusionApplication/tests/test_ConvectionDiffusionApplication.py]: Should be fine now.

**Comment by philbucher** [comment]
from my side looks good now, nice work :+1:

**Comment by rubenzorrilla** [comment]
After the fixes in the nightly build it should be fine.

**Review by philbucher — APPROVED** [review]

**Pull request merged**